packages/ts/frontend/src/Connect.ts

```diff
@@ -304,13 +340,31 @@ export class ConnectClient {
     const csrfHeaders = globalThis.document ? getCsrfTokenHeadersForEndpointRequest(globalThis.document) : {};
     const headers: Record<string, string> = {
       Accept: 'application/json',
-      'Content-Type': 'application/json',
       ...csrfHeaders,
     };
 
+    const [paramsWithoutFiles, files] = extractFiles(params ?? {});
+    let body;
+
+    if (files.size > 0) {
+      // in this case params is not undefined, otherwise there would be no files
+      body = new FormData();
+      body.append(
+        bodyPartName,
+        JSON.stringify(paramsWithoutFiles, (_, value) => (value === undefined ? null : value)),
+      );
+
+      for (const [path, file] of files) {
+        body.append(path, file);
+      }
+    } else {
+      headers['Content-Type'] = 'application/json';
+      body =
+        params !== undefined ? JSON.stringify(params, (_, value) => (value === undefined ? null : value)) : undefined;
```

Comment:
We can make it a bit more readable.

```suggestion
        if (params) {
          body = JSON.stringify(params, (_, value) => (value === undefined ? null : value))
        }
```